### PR TITLE
Keg-CLI tap docker tasks

### PR DESCRIPTION
**Pull request by lancetipton**
**Ticket**: [ZEN-456](https://jira.simpleviewtools.com/browse/ZEN-456)

[Semver-Status] minor

> ### Notes
> * This pr is dependent on two other PR's
>   * [tap-events-force](https://github.com/simpleviewinc/tap-events-force/pull/99)
>   * [keg-consumer](https://github.com/simpleviewinc/keg-test-consumer/pull/10)
> * Those PRs should be reviewed along with this one

## Context

* To add some tasks to the Keg-Cli for Taps

## Goal

* Add some docker tasks in the Keg-Cli for taps

## Updates

* `repos/keg-cli/src/tasks/docker/copy.js`
  * Minor update to ensure injected apps container can be used
* `repos/keg-cli/src/tasks/tap/docker/*`
  * Added some tap docker tasks
  * This ensure tap can properly use docker tasks not managed by docker-compose

## Testing

* Run the tests from [this keg-consumer PR](https://github.com/simpleviewinc/keg-test-consumer/pull/10)
  * All changes in this repo, are used within the PR's testing steps

### CODE REVIEW

File: repos/keg-cli/src/tasks/tap/docker/exec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    alias: [ 'ex' ],
+    inject: true,
+    action: exec,
+    locationContext: DOCKER.LOCATION_CONTEXT.REPO,
+    description: `Run a command in a tap container`,
+    example: 'keg tap docker exec',
+    options: {
+      cmd: {
+        description: 'Docker container command to run. Default ( /bin/bash )',
+        example: 'keg tap exec --cmd test',
```
Comment (daniel112) [inline repos/keg-cli/src/tasks/tap/docker/exec.js:37]: typo in the example. should be `keg tap docker exec` right?

File: repos/keg-cli/src/tasks/tap/docker/copy.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      source: {
+        alias: [ 'src',  ],
+        allowed: [ 'docker', 'host' ],
+        description: 'Source of the files to be copied.\nCopy from host: "local" || "host"\nCopy from docker: "docker" || "remote"',
+        default: 'docker'
+      },
+      local: {
+        alias: [ 'loc' ],
+        description: 'Local path to copy files to and from',
+        example: 'keg tap docker create --local ~/keg/keg-core',
```
Comment (daniel112) [inline repos/keg-cli/src/tasks/tap/docker/copy.js:46]: `keg tap docker copy`?

File: repos/keg-cli/src/tasks/tap/docker/copy.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        enforced: true,
+      },
+      remote: {
+        alias: [ 'rem' ],
+        description: 'Remote path to copy files to and from',
+        example: 'keg tap docker --remote keg/keg-core',
```
Comment (daniel112) [inline repos/keg-cli/src/tasks/tap/docker/copy.js:52]: `keg tap docker copy`?

**Review by daniel112 — APPROVED** [review]
tested on the consumer tests instructions. 
just some minor non blocking comments

File: repos/keg-cli/src/tasks/tap/docker/exec.js (hunk 1 above)
Comment (daniel112) [reply repos/keg-cli/src/tasks/tap/docker/exec.js:37]: updated

File: repos/keg-cli/src/tasks/tap/docker/copy.js (hunk 2 above)
Comment (daniel112) [reply repos/keg-cli/src/tasks/tap/docker/copy.js:46]: updated

File: repos/keg-cli/src/tasks/tap/docker/copy.js (hunk 3 above)
Comment (daniel112) [reply repos/keg-cli/src/tasks/tap/docker/copy.js:52]: updated

**Review by mpcarolin — APPROVED** [review]
Looks good

**Pull request merged**